GitOut/Features/IO/FileName.cs

```diff
@@ -6,6 +6,9 @@ namespace GitOut.Features.IO
 {
     public class FileName
     {
+        // the tag is used to differentiate with the other ctor and since it's protected will only (at the moment) be used by RelativeFileName
+        protected FileName(string filename, RelativeTag _) => Name = filename;
```

Comment:
We still want to validate the incoming file name to not contain any invalid characters.
If this is relative we can also validate that the filename is not rooted.

Reply:
Would it make sense to have FileName as is, and use a `GetRelativePath` method or something instead of using a subclass, or what exactly is the subclass accomplishing atm?

Reply:
Basically allowing FileName to still validate for directory characters and allowing a relative path with filename to be passed to a method expecting a FileName.
We can merge it, but we'd lose the validation. 🤷
It was a pretty quick fix, so I'm open to other ways to solve it 😄

Reply:
There is already a RelativeDirectoryPath, can we try to combine the two instead, since a FileName cannot really be "relative"? :thinking: